pkg/runtime/runtime-manifest.go

```diff
@@ -202,13 +239,31 @@ func (m *RuntimeManifest) buildSourceData() (map[string]interface{}, error) {
 		if err != nil {
 			return nil, err
 		}
-		m.sensitiveValues = append(m.sensitiveValues, val)
+		m.setSensitiveValue(val)
 		creds[pe] = val
 	}
 
 	bun["outputs"] = m.outputs
-	for _, output := range m.outputs {
-		m.sensitiveValues = append(m.sensitiveValues, output)
+	// Iterate through the runtime manifest's step outputs and mask by default
+	for _, stepOutput := range m.outputs {
+		// TODO: check if output declared sensitive or not; currently we default always sensitive
```

Comment:
Do we need to create a follow-up issue to track this todo?

Reply:
Good call; I created https://github.com/deislabs/porter/issues/855 and added a link below this comment.